### PR TITLE
Adding isDeleted property

### PR DESCRIPTION
**Pull request by pzaichkina**
# Description

Added isDeleted property to:
* [Querying](https://docs.codat.io/using-the-api/querying)
* [Refresh company data](https://docs.codat.io/using-the-api/queueing-data-syncs)
* [Using modified dates](https://docs.codat.io/using-the-api/modified-dates)

## Type of change

Please delete options that are not relevant.

- [X] New document(s)/updating existing

### CODE REVIEW

File: docs/using-the-api/modified-dates.md [hunk 1]
```diff
@@ -67,7 +67,8 @@ This query retrieves invoices with issue dates that are greater than twelve mont
 - The `sourceModifiedDate` may not be populated and returned as `null` when:
   - Pulling attachments, balance sheets, company information, or profit & loss reports
   - The integration platform does not provide modification date information for a given data type
-  - The record has been deleted from the source platform, but Codat doesn't have a record of when the deletion occurred. Void records may also be identified in the same way if the platform uses soft deletes.
+  - The record has been deleted from the source platform, but Codat doesn't have a record of when the deletion occurred. Void records may also be identified in the same way if the platform uses soft deletes.
```
Comment (dcoplowe) [inline docs/using-the-api/modified-dates.md:70]: What are we trying to say here?

File: docs/using-the-api/querying.md [hunk 2]
```diff
@@ -240,6 +240,43 @@ var info = response.Data;
...
+
+<Tabs>
+<TabItem value="http" label="HTTP">
+
+```http
+GET /companies/{companyId}/data/invoices?metadata.isDeleted%21%3dtrue
+```
+
+</TabItem>
+<TabItem value="javascript" label="Javascript">
```
Comment (dcoplowe) [inline docs/using-the-api/querying.md:257]: Can we use the speakeasy library?

Comment (dcoplowe) [reply docs/using-the-api/querying.md:257]:
```
import { CodatAccounting } from "@codat/accounting";
import { ListInvoicesResponse } from "@codat/accounting/dist/sdk/models/operations";
import { BilledToType1, InvoiceStatus } from "@codat/accounting/dist/sdk/models/shared";

const sdk = new CodatAccounting({
  security: {
    authHeader: "Basic BASE_64_ENCODED(API_KEY)",
  },
});

sdk.invoices.list({
  companyId: "8a210b68-6988-11ed-a1eb-0242ac120002",
  orderBy: "-modifiedDate",
  page: 1,
  pageSize: 100,
  query: "molestiae",
}).then((res: ListInvoicesResponse) => {
  if (res.statusCode == 200) {
    // handle response
  }
});
```

Comment (pzaichkina) [reply docs/using-the-api/querying.md:257]: we can, but which one of the examples does this replace?

File: docs/using-the-api/modified-dates.md [hunk 3]
```diff
@@ -67,7 +67,8 @@ This query retrieves invoices with issue dates that are greater than twelve mont
 - The `sourceModifiedDate` may not be populated and returned as `null` when:
   - Pulling attachments, balance sheets, company information, or profit & loss reports
```
[the remark this answers is not in the document]
Comment (pzaichkina) [reply docs/using-the-api/modified-dates.md:68]: attachments do have this property. for the other ones, if they truly don't, let's remove it then

File: docs/using-the-api/modified-dates.md (hunk 1 above)
Comment (pzaichkina) [reply docs/using-the-api/modified-dates.md:70]: well that's another scenario of when the value can be null. if the record was deleted in the accounting platform, but we don't know exactly when

File: docs/using-the-api/modified-dates.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (dcoplowe) [reply docs/using-the-api/modified-dates.md:68]: balance sheets, company information, and profit & loss don't have sourceModifiedDate.

File: docs/using-the-api/modified-dates.md (hunk 1 above)
Comment (dcoplowe) [reply docs/using-the-api/modified-dates.md:70]:
```suggestion
  - If a record has been deleted from the source platform, Codat doesn't have a record of when the deletion occurred.
  - Platforms that soft delete records can use the `isDeleted` metadata to identify void records.
```

File: docs/using-the-api/querying.md (hunk 2 above)
Comment (dcoplowe) [reply docs/using-the-api/querying.md:257]: This replaces the javascript one.

**Review by mcclowes — APPROVED** [review]

**Pull request merged**